spec.bs

```diff
@@ -687,6 +696,55 @@ Moreover, each {{SharedStorageWorklet}}'s [=global scopes|list of global scopes=
     This rationale also applies to the handling for user preferences error for {{SharedStorageWorklet/selectURL()}} and {{SharedStorageWorklet/run()}}.
     </div>
 
+  After the step "Let <var ignore=''>addedSuccessfully</var> be false", we need to include the following step:
+
+    4. If |this| is of type {{SharedStorageWorklet}}, [=SharedStorageWorklet/has cross-origin data origin=] is true, and [=SharedStorageWorklet/data origin=] is not `"script-origin"`:
+        1. [=Assert=] |pendingTasks| is 1.
+        1. Set |pendingTasks| to 2.
+        1. [=Queue a global task=] on the [=networking task source=] given <var ignore=''>workletGlobalScope</var> to perform the following steps:
+            1. Let |customOriginUrl| be the result of running a [=URL parser=] on [=SharedStorageWorklet/data origin=].
+            1. [=Assert=] |customOriginUrl| is a valid [=/URL=].
+            1. Set |customOriginUrl|'s [=url/path=] to ≪".well-known", "shared-storage", "trusted-origins"≫.
+            1. Let |request| be a new [=/request=] whose [=request/URL=] is |customOriginUrl|, [=request/mode=] is `"cors"`, [=request/referrer=] is `"client"`, [=request/destination=] is `"json"`, [=request/initiator type=] is `"script"`, and [=request/client=] is |outsideSettings|.
+            1. [=Fetch=] |request| with [=fetch/processResponseConsumeBody=] set to the following algorithm, given [=/response=] |response| and null, failure or a [=/byte sequence=] |bodyBytes|:
+                1. If any of the following are true:
+                    * |bodyBytes| is null or failure; or
+                    * |response|'s [=response/status=] is not an [=ok status=],
+
+                       then:
+                       1. Set |pendingTasks| to −1.
+                       1. [=Reject=] |promise| with an "TypeError" DOMException.
+                       1. Abort these steps.
+                1. Let |mimeType| be the result of [=extracting a MIME type=] from |response|'s [=response/header list=].
+                1. If |mimeType| is not a [=JSON MIME type=], then:
+                    1. Set |pendingTasks| to −1.
+                    1. [=Reject=] |promise| with an "TypeError" DOMException.
+                    1. Abort these steps.
+                1. Let |sourceText| be the result of [=UTF-8 decoding=] |bodyBytes|.
+                1. Let |parsed| be the result of [=parsing a JSON string to an Infra value=] given |sourceText|.
+                1. If |parsed| is not a [=list=] or if |parsed| is [=list/empty=], then:
+                    1. Set |pendingTasks| to −1.
+                    1. [=Reject=] |promise| with an "TypeError" DOMException.
+                    1. Abort these steps.
+                1. Let |doesMatch| be false.
+                1. For each |item| of |parsed|:
+                    1. If |item| is not an [=ordered map=], then continue.
```

Comment:
Why continue instead of throwing an error here?  Should we not be strict about the format of the data we accept?

Reply:
Changed to reject.